src/Microsoft.IdentityModel.Tokens/ClaimsIdentity/ClaimsProvider.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Security.Claims;
+
+namespace Microsoft.IdentityModel.Tokens
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public abstract class ClaimsProvider : SecurityToken
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="type"></param>
+        /// <returns></returns>
+        public virtual Claim GetPayloadClaim(string type) => null;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="type"></param>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        public virtual bool HasPayloadClaim(string type, string value) => false;
```

Comment:
This naming makes sense for JWT, but not for SAML

Reply:
The naming is sort of temporary for now. We have similarly named methods that perform old logic; so tried to make these stand out for the ease of refactoring for now.